src/devices/schou_72543_rain.c

```diff
@@ -82,12 +81,13 @@ static int schou_72543_rain_decode(r_device *decoder, bitbuffer_t *bitbuffer)
 
     uint8_t b = bitbuffer->bb[row];
 
-    uint8_t micSum = b[7]; // Checksum as read
-    uint8_t calSum = 0;    // Checksum as calculated
-
-    for(i = 0; i<7; i++){
-        calSum = calSum + b[i];
-    }
+    uint8_t micSum = b[7];              // Checksum as read
+    int    cal_sum = add_bytes(b, 7);   // Checksum as calculated
+    //uint8_t calSum = 0;               // Checksum as calculated
+    //
+    //for (i = 0; i<7; i++){
+    //    calSum = calSum + b[i];
+    //}
 
     if (micSum != calSum) {
```

Comment:
this now needs `if (micSum != calSum & 0xff) {`

Reply:
moved to calSum